plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -153,13 +153,22 @@ private static org.apache.iceberg.types.Type toIcebergTypeInternal(Type type, Op
         if (type instanceof DateType) {
             return Types.DateType.get();
         }
-        if (type.equals(TIME_MICROS)) {
+        if (type instanceof TimeType timeType) {
+            if (timeType.getPrecision() > 6) {
+                throw new TrinoException(NOT_SUPPORTED, format("Time precision (%s) not supported for Iceberg. Use \"time(6)\" instead.", timeType.getPrecision()));
+            }
             return Types.TimeType.get();
         }
-        if (type.equals(TIMESTAMP_MICROS)) {
+        if (type instanceof TimestampType timestampType) {
+            if (timestampType.getPrecision() > 6) {
```

Comment:
Considering that the table `foo` has a column `col` of type `timestamp(6)`, why would we allow statements like the following:
```
ALTER TABLE foo ALTER COLUMN col SET DATA TYPE timestamp(3)
```
?

Reply:
My understanding is that even when internally iceberg stores time values using microseconds precision. We could still allow any precision in Trino.

However I agree this goes beyond the scope of this PR, I will restore the existing behaviour.